src/routes/safe/components/Transactions/TxList/hooks/useLastQueuedTxNonce.ts

```diff
@@ -0,0 +1,19 @@
+import { useSelector } from 'react-redux'
+import { nextTransactions, queuedTransactions } from 'src/logic/safe/store/selectors/gatewayTransactions'
+
+const useLastQueuedTxNonce = (): number | undefined => {
+  const queuedTxs = useSelector(queuedTransactions)
+  const nextTxs = useSelector(nextTransactions)
+  const queuedNonces = queuedTxs && Object.keys(queuedTxs)
+  const nextNonces = nextTxs && Object.keys(nextTxs)
```

Comment:
Have you seen Liliya's comment? She suggested to use `safeInfo.nonce` from the store. Not sure how often it's updated tho.

Reply:
The `safeNonce` that Liliya mentions in the comment holds the next Transaction Nonce that will be executed from the safe (same value as the tx history + 1)
Here we are fetching the queued nonces.